packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -1146,6 +1217,94 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
             .toList();
     }
 
+    onBulkUpdateFormDataChange = (pendingBulkFormData?: any): void => {
+        this.setState({
+            pendingBulkFormData,
+        });
+    };
+
+    onTabChange = (event: SyntheticEvent<TabContainer, Event>) => {
+        const { bulkUpdateProps } = this.props;
+        const { activeEditTab, pendingBulkFormData } = this.state;
+        const newTabKey: EditableGridTabs = event as any; // Crummy cast to make TS happy
+
+        if (newTabKey === EditableGridTabs.Grid && activeEditTab === EditableGridTabs.BulkUpdate) {
+            this.bulkUpdate(pendingBulkFormData).then(() => {
+                this.setState(() => ({
+                    pendingBulkFormData: undefined,
+                    activeEditTab: newTabKey,
+                }));
+            });
+        } else if (newTabKey === EditableGridTabs.BulkUpdate && activeEditTab === EditableGridTabs.Grid) {
+            if (bulkUpdateProps?.onClickBulkUpdate) {
+                bulkUpdateProps
+                    .onClickBulkUpdate(this.state.selected)
+                    .then(canEdit => {
+                        if (!canEdit) return;
+
+                        this.setState(() => ({ activeEditTab: newTabKey }));
+                    })
+                    .catch(error => {
+                        console.error(error);
```

Comment:
For the current usages, I had the consumer/caller render the error msg when onClickBulkUpdate errors, because they have better knowledge when to clear out the error.